examples/client-benchmark/main.cpp

```diff
@@ -0,0 +1,460 @@
+	const string label = "benchmark";
+	cout << "Creating DataChannel with label \"" << label << "\"" << endl;
+	auto dc = pc->createDataChannel(label);
+
+	// Set Buffer Size
+	dc->setBufferedAmountLowThreshold(bufferSize);
+
+	dc->onOpen([id, wdc = make_weak_ptr(dc)]() {
+		cout << "DataChannel from " << id << " open" << endl;
+		if (noSend)
+			return;
+
+		if (enableThroughputSet)
+			return;
+
+		if (auto dcLocked = wdc.lock()) {
+			try {
+				while (dcLocked->bufferedAmount() <= bufferSize) {
+					dcLocked->send(messageData);
+					sentSize += messageData.size();
+				}
+			} catch (const std::exception &e) {
+				std::cout << "Send failed: " << e.what() << std::endl;
+			}
+		}
+	});
+
+	dc->onBufferedAmountLow([wdc = make_weak_ptr(dc)]() {
+		if (noSend)
+			return;
+
+		if (enableThroughputSet)
+			return;
+
+		auto dcLocked = wdc.lock();
+		if (!dcLocked)
+			return;
+
+		// Continue sending
+		try {
+			while (dcLocked->isOpen() && dcLocked->bufferedAmount() <= bufferSize) {
+				dcLocked->send(messageData);
+				sentSize += messageData.size();
+			}
+		} catch (const std::exception &e) {
+			std::cout << "Send failed: " << e.what() << std::endl;
+		}
+	});
+
+	dc->onClosed([id]() { cout << "DataChannel from " << id << " closed" << endl; });
+
+	dc->onMessage([id, wdc = make_weak_ptr(dc)](variant<binary, string> data) {
+		if (holds_alternative<binary>(data))
+			receivedSize += get<binary>(data).size();
+	});
+
+	dataChannelMap.emplace(id, dc);
+
+	const int duration = params.durationInSec() > 0 ? params.durationInSec() : INT32_MAX;
+	cout << "Benchmark will run for " << duration << " seconds" << endl;
+
+	int printCounter = 0;
+	int printStatCounter = 0;
+	steady_clock::time_point printTime = steady_clock::now();
+	steady_clock::time_point stepTime = steady_clock::now();
+	// Byte count to send for every loop
+	int byteToSendOnEveryLoop = throughtputSetAsKB * stepDurationInMs;
+	for (int i = 1; i <= duration * STEP_COUNT_FOR_1_SEC; ++i) {
+		this_thread::sleep_for(milliseconds(stepDurationInMs));
+		printCounter++;
+
+		if (enableThroughputSet) {
+			float dataCountMultiplier =
+			    duration_cast<microseconds>((steady_clock::now() - stepTime)).count() /
+			    (1000.0 * stepDurationInMs);
+
+			stepTime = steady_clock::now();
+
+			int byteToSendThisLoop = static_cast<int>(byteToSendOnEveryLoop * dataCountMultiplier);
+
+			binary tempMessageData(byteToSendThisLoop);
+			fill(tempMessageData.begin(), tempMessageData.end(), std::byte(0xFF));
+
+			if (dc->isOpen() && dc->bufferedAmount() <= bufferSize * byteToSendOnEveryLoop) {
+				dc->send(tempMessageData);
+				sentSize += tempMessageData.size();
+			}
+		}
+
+		if (printCounter >= STEP_COUNT_FOR_1_SEC) {
+			unsigned long _receivedSize = receivedSize.exchange(0);
+			unsigned long _sentSize = sentSize.exchange(0);
+			float dataCountMultiplier =
+			    duration_cast<milliseconds>((steady_clock::now() - printTime)).count() /
+			    (1000.0); // sec
```

Comment:
It looks quite convoluted to compute this `dataCountMultiplier`. Why not compute an actual `elaspedTime` of type `seconds` and just divide the size by it? You could do the same in the sending thread.

Reply:
Thanks for the hint.